### PR TITLE
1122 chemical page products

### LINKED ISSUE #1122: display products on chemical summary page
As a user,
I want to easily link to all products including a particular chemical and in a particular PUC
So that I can investigate data relevant to my PUC and chemical of interest

http://factotum.epa.gov/chemical/DTXSID7020637/

Currently, the /chemical/ page includes a bubble plot/table which displays the number of products in each PUC containing the chemical of interest, but there is no way to access a list of those products.

Definition of Done:
- Ability to see, on /chemical/ page, products in a PUC that contain the chemical of interest (one option for implementation is that user can click on a PUC in either the bubble plot or the summary table next to the bubble plot and then a list of the relevant products appears as a table below the PUC plot; OR user always sees a table listing all products containing that chemical, and can filter the table by the PUC they want to look at)
- In presentation/table of products, user would see product name associated document title, and PUC. Product name should be hyperlink to the /product/ page, document title should be hyperlink to /datadocument/ page, and PUC should be hyperlink to associated /PUC/ page

### PR DESCRIPTION
**Pull request by zachmunzenrider**
Closes #1122 

I've been developing this ticket against prod data, at http://127.0.0.1:8000/chemical/DTXSID6026296, which is water. It's got a lot of products and is a bit sluggish on this specific chemical, but I think it's peppy enough. We might be able to refine the underlying Ajax in the future to speed it up.

I also expect Kathie to have additional ideas about how the functionality of the filtering. For now, when you click on a PUC filter, it applies the filter to both the Documents and the Products tables.

This ticket leaves the chemical page in a transitional state--I pulled out some of the lovely CSS toggling, and the tables are stacked on the page in a way that requires lots of scrolling and generally isn't elegant. We had additional tickets in the queue to address the redesign of this page, and the changes I've made here get us closer to that goal, even if the page doesn't look as nice in the interim.

### CODE REVIEW

**Comment by michael-barbour** [comment]
This ticket is close to being reviewer approved.  There are no tests currently (addressed in above review).  These pending tests, as well as the changes I requested to be addressed, are currently acting as blockers for review.  Zach is currently on vacation, but attempting to get the tests written in order to make it into the current sprint.

File: templates/chemicals/chemical_detail.html [hunk 1]
```diff
@@ -2,129 +2,129 @@
...
-                        </div>
-                        {% if keyset.keywords|join:" , "|length > 45 %}
-                        <div id="pid-{{keyset.presence_id}}"
-                            class="collapse" 
-                            aria-labelledby="keyCard" 
-                            data-parent="#keyAccordion">
-                            <div class="card-body py-0">
-                                <b>{ {{ keyset.keywords|join:" , " }} }</b>
+                                </div>
+                                {% if keyset.keywords|join:" , "|length > 45 %}
```
[the remark this answers is not in the document]
Comment (michael-barbour) [reply templates/chemicals/chemical_detail.html]: Number were changed to match in latest commit.  The comment regarding the limit change above still stands.

File: templates/chemicals/chemical_detail.html [hunk 2]
```diff
@@ -2,129 +2,129 @@
...
-                                        data-target="#pid-{{keyset.presence_id}}" aria-expanded="true" 
+                                        {% if keyset.keywords|join:' , '|length <= 135 %}
+                                        disabled
+                                        {% endif %}
+                                        type="button"
+                                        data-toggle="collapse"
+                                        data-target="#pid-{{ keyset.presence_id }}" aria-expanded="true"
                                         aria-controls="collapseOne">
-                                    <b>{ {{ keyset.keywords|join:" , "|truncatechars:45 }} }</b>
+                                    <b>{ {{ keyset.keywords|join:" , "|truncatechars:135 }} }</b>
```
[the remark this answers is not in the document]
Comment (zachmunzenrider) [reply templates/chemicals/chemical_detail.html]: Michael, I changed the truncate limit (in all 3 spots) to 80. This should address the wraparound issue you were seeing. It still wraps around when you do the dropdown, but I think that's ok for the time being. Check this out (with the dropdown enabled):

image

[the remark this answers is not in the document]
Comment (michael-barbour) [reply templates/chemicals/chemical_detail.html]: This looks much better.  Lines don't word wrap while I am fullscreen.  

There is still an issue with reducing the side causing a word wrap but that is outside the scope of the ticket.

File: dashboard/tests/integration/test_puc_detail.py [hunk 3]
```diff
@@ -68,7 +68,7 @@ def test_puc_product_datatable(self):
         )
 
         # Chemicals
-        chem_btn = self.browser.find_element_by_id("quimicos")
+        chem_btn = self.browser.find_element_by_id("chemicals")
```
Comment (michael-barbour) [inline dashboard/tests/integration/test_puc_detail.py]: This was named quimicos

File: static/js/dashboard/puc_detail.js [hunk 4]
```diff
@@ -50,11 +50,11 @@ $(document).ready(function () {
             stateSave: true,
             ajax: "/d_json/?puc=" + puc,
         });
-    })
+    });
 
-    $("#quimicos").on("click", function(){
-        $("div[id$='table-div']").hide()
-        $("#chem-table-div").show()
+    $("#chemicals").on("click", function(){
```
Comment (michael-barbour) [inline static/js/dashboard/puc_detail.js]: This was also named quimicos

File: templates/puc/puc_detail.html [hunk 5]
```diff
@@ -105,31 +105,31 @@ <h4>Linkages to Other Taxonomies</h4>
...
                     <b>Documents</b>
                 </a>
             </li>
             <li class="col-lg-4 nav-item h-nav border rounded p-0">
                 <a class="nav-link text-center" 
                    data-toggle="pill" 
-                   href="#chem-table-div"
-                   id="quimicos">
+                   href="#chemical-table-div"
+                   id="chemical-tab-header">
```
Comment (michael-barbour) [inline templates/puc/puc_detail.html]: This was named quimicos but was translated to chemical-tab-header.  It needs to be the same as the other two functions.  One in the selenium test `test_puc_detail.py` and one on the JS that drives this page `puc_detail.js`

**Pull request merged**